packages/medusa/src/controllers/customers.ts

```diff
@@ -0,0 +1,29 @@
+import { omit, pickBy } from "lodash"
+import { AdminGetCustomersParams, Customer } from ".."
+import { CustomerService } from "../services"
+import { FindConfig } from "../types/common"
+
+const listAndCount = async (
+  customerService: CustomerService,
+  queryparams: AdminGetCustomersParams
+): Promise<any> => {
+  let expandFields: string[] = []
+  if (queryparams.expand) {
+    expandFields = queryparams.expand.split(",")
+  }
+
+  const listConfig: FindConfig<Customer> = {
+    relations: expandFields,
+    skip: queryparams.offset,
+    take: queryparams.limit,
+  }
+
+  const filterableFields = omit(queryparams, ["limit", "offset", "expand"])
+
+  return await customerService.listAndCount(
+    pickBy(filterableFields, (val) => typeof val !== "undefined"),
+    listConfig
+  )
+}
```

Comment:
I think we should try to standardize the inputs to "controller" functions; could be something like this:

- `scope`, `queryParams`, `payload`, `context`; so when called from the endpoint handler you'd pass 

```
listCustomers(req.scope, req.query, req.body, { ... we'll figure out what should go in here })
```

- I also think that the return type of this should correspond to the `[domain]Res` type i.e. for this it should be:
```
{
  customers: [...],
  count,
  limit,
  offset
}
```